x-pack/plugins/ml/common/util/job_utils.ts

```diff
@@ -143,6 +143,15 @@ export function isSourceDataChartableForDetector(job: CombinedJob, detectorIndex
       if (isPopulatedObject(aggs)) {
         const aggBucketsName = getFirstKeyInObject(aggs);
         if (aggBucketsName !== undefined) {
+          // We current don't support aggregations that reference other nested aggregations
+          // e.g. aggregation that reference another nested term aggregation
+          if (
+            Object.keys(aggs[aggBucketsName]).some(
+              (key) => key !== 'date_histogram' && key !== 'aggs' && key !== 'aggregations'
```

Comment:
shall we create a function for this case, e.g. `isRefAggregation` or something similar?

Reply:
I ended up creating `isUnsupportedAggType` because I think it's a bit clearer [`b6f3246` (#98510)](https://github.com/elastic/kibana/pull/98510/commits/b6f3246828c2311053ddd0a56ce9684b941bb895) what we can and cannot currently support.